prime-router/src/test/kotlin/azure/ActionHistoryTests.kt

```diff
@@ -0,0 +1,146 @@
+            ReportFunction.Options.CheckConnections, mapOf(),
+            listOf<ResultDetail>(),
+            listOf<ResultDetail>(), report1
+        )
+        val actionHistory1 = ActionHistory(TaskAction.receive)
+        actionHistory1.trackExternalInputReport(incomingReport)
+        assertNotNull(actionHistory1.reportsReceived[report1.id])
+        val reportFile = actionHistory1.reportsReceived[report1.id] !!
+        assertEquals(reportFile.schemaName, "one")
+        assertEquals(reportFile.schemaTopic, "test")
+        assertEquals(reportFile.sendingOrg, "myOrg")
+        assertEquals(reportFile.sendingOrgClient, "myClient")
+        assertNull(reportFile.receivingOrg)
+
+        // not allowed to track the same report twice.
+        assertFails { actionHistory1.trackExternalInputReport(incomingReport) }
+
+        // must pass a valid report.   Here, its set to null.
+        val incomingReport2 = ReportFunction.ValidatedRequest(
+            ReportFunction.Options.CheckConnections, mapOf(),
+            listOf<ResultDetail>(),
+            listOf<ResultDetail>(), null
+        )
+        assertFails { actionHistory1.trackExternalInputReport(incomingReport2) }
+    }
+
+    @Test
+    fun `test trackCreatedReport`() {
+        val event1 = ReportEvent(Event.EventAction.TRANSLATE, UUID.randomUUID(), OffsetDateTime.now())
+        val schema1 = Schema(name = "schema1", topic = "topic1", elements = listOf())
+        val report1 = Report(schema1, listOf(), sources = listOf(ClientSource("myOrg", "myClient")))
+        val valReq1 = ReportFunction.ValidatedRequest(
+            ReportFunction.Options.CheckConnections, mapOf(),
+            listOf<ResultDetail>(),
+            listOf<ResultDetail>(), report1
+        )
+        val org =
+            Organization(
+                name = "myOrg",
+                description = "blah blah",
+                clients = listOf(),
+                services = listOf(
+                    OrganizationService("myService", "topic", "schema")
+                )
+            )
+        val orgSvc = org.services[0]
+        val actionHistory1 = ActionHistory(TaskAction.receive)
+
+        actionHistory1.trackCreatedReport(event1, report1, orgSvc)
+
+        assertNotNull(actionHistory1.reportsOut[report1.id])
+        val reportFile = actionHistory1.reportsOut[report1.id] !!
+        assertEquals(reportFile.schemaName, "schema1")
+        assertEquals(reportFile.schemaTopic, "topic1")
+        assertEquals(reportFile.receivingOrg, "myOrg")
+        assertEquals(reportFile.receivingOrgSvc, "myService")
+        assertNull(reportFile.sendingOrg)
+        assertEquals(reportFile.itemCount, 0)
+
+        // not allowed to track the same report twice.
+        assertFails { actionHistory1.trackCreatedReport(event1, report1, orgSvc) }
+    }
+
+    /**
+     * todo Figure out how to make this test work.
+     * What I'd really like to do is confirm that two sql inserts were generated,
```

Comment:
I need some help with this.  Unable to figure out how to create a mock transaction.  Further, unable to figure out how to make this test anything useful.